### PR TITLE
Add jupyter to the docker image

### PR DESCRIPTION
**Pull request by vmalarcon**

### CODE REVIEW

File: docker/standalone/cpu/Dockerfile [hunk 1]
```diff
@@ -33,11 +34,13 @@ RUN git clone -b ${CLONE_TAG} --depth 1 https://github.com/BVLC/caffe.git . && \
     for req in $(cat python/requirements.txt) pydot; do pip install $req; done && \
     mkdir build && cd build && \
     cmake -DCPU_ONLY=1 .. && \
-    make -j"$(nproc)"
+    make all -j"$(nproc)" && \
+    make pycaffe
```
Comment (lukeyeager) [inline docker/standalone/cpu/Dockerfile]: CMake builds pycaffe automatically for me when Python is detected. Does it not for you?

**Comment by flx42** [comment]
You are only modifying the CPU Dockerfile here, please modify the template file and regenerate the two Dockerfiles.

**Comment by flx42** [comment]
Also, you need a minimum jupyter configuration file to set `open_browser = False` and to modify the IP you are listening on.

**Comment by flx42** [comment]
I said a **minimum** configuration file :)
And you still don't need to modify the `make` command.

**Comment by flx42** [comment]
I think the following is enough:
```
c.NotebookApp.ip = '*'
c.NotebookApp.port = 8888
c.NotebookApp.open_browser = False
```
Also, ping @elezar

**Comment by elezar** [comment]
Thanks for the addition @vmalarcon!

If the consensus is that the images should include Jupyter, then I agree with @flx42 that the config file should be kept minimal, and in this case, we could consider creating it in the Dockerfile:
```
RUN echo "c.NotebookApp.ip = '*'" > configfile
RUN echo "c.NotebookApp.port = 8888" >> configfile
RUN echo "c.NotebookApp.open_browser = False" >> configfile
```
(I generally specify these on the command line when starting a notebook server)
but I don't know whether this is in keeping with Docker best practices and separating the config into a separate file is probably better.

Another option is to have the jupyter version in a separate docker image? I use such an image locally for building and testing caffe. I do this using the following files:
- [Dockerfile.txt](https://github.com/BVLC/caffe/files/258684/Dockerfile.txt)
- [start_caffe_docker.sh.txt](https://github.com/BVLC/caffe/files/258683/start_caffe_docker.sh.txt)
(note that the `.txt` extension should be removed in each case)

**Comment by vmalarcon** [comment]
Thank you guys for the comments/help @flx42 @elezar! 

I like @elezar 's idea of using a separate Dockerfile for jupyter and inherit from caffe:[g|c]pu. I think I'll do that on my end. So I'll close this PR. Actually, I never intended this to be merged to the main repo.

**Comment by vmalarcon** [comment]
https://github.com/vmalarcon/caffe/pull/1

**Comment by elezar** [comment]
Glad we could help!

**Pull request closed without merging**